### PR TITLE
Comment Moderation Bar: add Approved, Spam, and Trash buttons.

### PR DESCRIPTION
**Pull request by ScoutHarris**
Ref: #17200 
Depends on: #17211

This adds the Approved, Spam, and Trash buttons to the moderation bar. Tapping the buttons will toggle the UI to indicate selected or not. No action is actually performed yet.

NOTE: The vertical dividers can show incorrectly when two buttons are selected. This isn't a real scenario since a comment can have only one state. It will be resolved once the comment state is applied.

To test:
- Enable the `newCommentDetail` feature.
- Go to My Site > Comments > Comment details.
- Verify all the buttons are in the unselected state.
- Tap a button. 
  - Verify the UI changes to the selected state.
  - Verify the vertical dividers on either side are hidden when selected, shown when unselected.

| Unselected | Selected |
|--------|-------|
| light_unselected | light_selected |
| dark_unselected | dark_selected |

## Regression Notes
1. Potential unintended areas of impact
N/A. Feature is incomplete and disabled.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A. Feature is incomplete and disabled.

3. What automated tests I added (or what prevented me from doing so)
N/A. Feature is incomplete and disabled.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Comments/CommentModerationBar.swift [hunk 1]
```diff
@@ -130,3 +177,15 @@ private extension UIButton {
     }
 
 }
+
+// MARK: - UIView Extension
+
+private extension UIView {
+    func configureAsDivider() {
+        backgroundColor = .textSubtle
```
Comment (dvdchr) [inline WordPress/Classes/ViewRelated/Comments/CommentModerationBar.swift:186]: Checked Zeplin, and I think this should be `.systemGray`.

Screen Shot 2021-09-24 at 18 46 33

It's really close to `textSubtle`, but the system gray looks a little bit more subtle / light. (Also, it was tough to click the divider 😅 .)

```suggestion
        backgroundColor = .systemGray
```

**Review by dvdchr — APPROVED** [review]
Left one minor comment. The rest looks good to me! 🙂 

> NOTE: The vertical dividers can show incorrectly when two buttons are selected. This isn't a real scenario since a comment can have only one state. It will be resolved once the comment state is applied.

👍 !

File: WordPress/Classes/ViewRelated/Comments/CommentModerationBar.swift (hunk 1 above)
Comment (ScoutHarris) [reply WordPress/Classes/ViewRelated/Comments/CommentModerationBar.swift:186]: I'll most likely have to update the colors anyway once Miklic reviews it, so I'll do it all in one go later. Also, I _could not_ select the divider in Zeplin! Thanks for the screenie!

**Comment by mattmiklic** [comment]
Thanks for the ping on the dark mode colors! I've updated Zeplin with a dark version of the Comment Detail screen and dark variants of each state of the moderation bar.

Screen Shot 2021-09-24 at 8 36 20 PM

**Pull request merged**